### DISCUSSION TITLE
UGS won't reconnect with GRBLHAL after unplugged and replugged in.

### DISCUSSION THREAD

**Discussion by jmonger1947** [winder/Universal-G-Code-Sender#2560] [Help]
I'm having a problem with A Teensy4.1 board programed with GRBLHAL. UGS can connect to the board if I don't power cycle the board (basically unplug/replug it). I can even go to the Arduino serial monitor and connect which it won't do after the power cycle. A strange thing is, I can go to TinyG part of UGS and connect with it then I can go to GRBL and it will connect correctly. I can disconnect and reconnect(in UGS) correctly as long as I don't unplug the board. So what's the difference between connecting with TinyG and GRBL and why does it continue to work after the initial connection with TinyG? I'm using Windows 10, ver 22H2, UGS ver 2.1.7, GRBLHAL version 1.1f.20210513, the Teensey is the "brains" for GRBLHAL Teensy 4.1 motion controller breakout board from Phil Barrett. The  records from the UGS console is included as an addition.
[GRBL problem.txt](https://github.com/user-attachments/files/16022684/GRBL.problem.txt)
Thanks to all for and help, speculation, or solutions.

**Comment by breiler** [comment]
Go to menu `Tools -> Options`. Click on the `UGS` tab, then `Sender Options`. Then change connection driver to `JSerialComm`.

![image](https://github.com/winder/Universal-G-Code-Sender/assets/8962024/a3546609-b0b9-4de5-ba4d-7e2af79925bc)


See if that helps.

**Reply by jmonger1947** [reply]
I tried your suggestion, it works exactly the same as the jssc connection driver, including the bit of using TinyG to innitiate a connection then going to GRBL.

**Reply by jmonger1947** [reply]
Does the TinyG use or do something different for connection?

**Reply by breiler** [reply]
Yes, I think that the difference is that it sends a reset command on connect. 

From your your log it looks like it is in HOLD state when you connect to it. Then after you've tried using the TinyG connection, the controller is reset and enters an IDLE state. 

I have the same hardware as you but a slightly newer version of grblHAL. I need to configure it in the same way to see if I can reproduce the problem. I know that JSSC driver can cause problems so I thought that it was worth trying first.
